core-client/src/main/java/org/glassfish/jersey/client/internal/HttpUrlConnector.java

```diff
@@ -314,10 +321,37 @@ protected void secureConnection(final JerseyClient client, final HttpURLConnecti
         }
     }
 
+    private static URI getProxyUriValue(Object proxy) {
+        if (proxy instanceof URI) {
+            return (URI) proxy;
+        } else if (proxy instanceof String) {
+            return URI.create((String) proxy);
+        } else {
+            throw new ProcessingException(LocalizationMessages.WRONG_PROXY_URI_TYPE(ClientProperties.PROXY_URI));
+        }
+    }
+
     private ClientResponse _apply(final ClientRequest request) throws IOException {
         final HttpURLConnection uc;
-
-        uc = this.connectionFactory.getConnection(request.getUri().toURL());
+        Proxy proxy = null;
+        Object proxyUri = request.getConfiguration().getProperties().get(ClientProperties.PROXY_URI);
```

Comment:
Would it be better to use `request.resolveProperty` to enable proxy settings in the filters?

Reply:
It is done in this way in other connectors too:
`request.getConfiguration().getProperties().get(ClientProperties.PROXY_URI);`
It looks the property could be a String or an URI. I am not sure why but I think it is safer to keep it consistent with other connectors.